src/extra_functions.rs

```diff
@@ -166,6 +167,85 @@ pub extern "C" fn TIME() -> i64 {
     dt.num_seconds_from_midnight() as i64 * 1e9 as i64 + dt.nanosecond() as i64
 }
 
+/// # Safety
+/// Uses raw pointers, inherently unsafe.
+#[allow(non_snake_case)]
+#[no_mangle]
+pub unsafe extern "C" fn TIME_TO_STRING_EXT(input: i64, dest: *mut u8) -> i32 {
+    let mut dest = dest;
+    let literals = parse_timestamp(input);
+    literals.iter().filter(|&it| it.0 != 0).for_each(|it| {
+        let buf = core::slice::from_raw_parts_mut(dest, DEFAULT_STRING_LEN);
+        write!(&mut *buf, "{}{}", it.0, it.1).unwrap();
+        let idx = buf.iter().position(|&c| c == 0).unwrap();
+        dest = dest.add(idx);
+    });
+
+    0
+}
+
+fn parse_timestamp<'a>(timestamp_nanos: i64) -> [(i64, &'a str); 7] {
+    let (nanos, timestamp_micros) = (timestamp_nanos % 1000, timestamp_nanos / 1000);
```

Comment:
I have considered this also.. It is not quite as simple though. TIME is a measurement of elapsed time and as such independent of a date. Arriving at the amount of elapsed days with the provided functions year(), month() and day() is not really straightforward, since day() only gives the day of the month. You'd have to check which month it is, add up all the days from previous months (which seems awkward, considering the days in any given month are inconsistent). 

I'll have a look at implementing this with a lookup table for days/month and adding up the days instead.

Reply:
have you looked into the chrono docs ? maybe there is something that could help
nevertheless your implementation works i just thought maybe there is some "slimer" solution to this

Reply:
I'll have another look. Anyway, I agree that my implementation wasn't very pretty. I have changed it to use the chrono functions for all values up until hours and then calculated the elapsed days manually.